dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/DynamicPureLoopCutting.java

```diff
@@ -122,22 +127,24 @@ private void insertSideEffectChecks(IterationData iter) {
 
         final List<Register> trackingRegs = new ArrayList<>();
         Event insertionPoint = iterInfo.getIterationEnd();
+        Type type = TypeFactory.getInstance().getPointerType();
         for (int i = 0; i < sideEffects.size(); i++) {
             final Event sideEffect = sideEffects.get(i);
             final Register trackingReg = thread.newRegister(
                     String.format("Loop%s_%s_%s", loopNumber, iterNumber, i),
-                    GlobalSettings.getArchPrecision());
+                    type);
```

Comment:
Okay, this shows that you just used the `PointerType` everywhere.
I won't keep pointing it out then, but please add something like `TypeFactory.getPointerSizedIntegerType` and use that instead.
Having `PointerType` for memory-unrelated values just feels too wrong to me.